Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+    //            AssetDatabase.CreateAsset(settings, k_MyCustomSettingsPath);
+    //            AssetDatabase.SaveAssets();
+    //        }
+    //        return settings;
+    //    }
+
+    //    internal static SerializedObject GetSerializedSettings()
+    //    {
+    //        return new SerializedObject(GetOrCreateSettings());
+    //    }
+    //}
+
+    //internal static class ScriptReferenceRetargettingSettingsIMGUIRegister
+    //{
+    //    [SettingsProvider]
+    //    public static SettingsProvider CreateMyCustomSettingsProvider()
+    //    {
+    //        SettingsProvider provider = new SettingsProvider("Project/ScriptReferenceRetargetter", SettingsScope.Project)
+    //        {
+    //            // By default the last token of the path is used as display name if no label is provided.
+    //            label = "Script Reference Retargetter Settings",
+    //            // Create the SettingsProvider and initialize its drawing (IMGUI) function in place:
+    //            guiHandler = (searchContext) =>
+    //            {
+    //                var settings = ScriptReferenceRetargettingSettings.GetSerializedSettings();
+    //                EditorGUILayout.PropertyField(settings.FindProperty("m_Number"), new GUIContent("My Number"));
+    //                EditorGUILayout.PropertyField(settings.FindProperty("m_SomeString"), new GUIContent("My String"));
+    //            },
+
+    //            // Populate the search keywords to enable smart search filtering and label highlighting:
+    //            keywords = new HashSet<string>(new[] { "Number", "Some String" })
+    //        };
+
+    //        return provider;
+    //    }
+    //}
+
+
+    public static class AssetScriptReferenceRetargeter
+    {
+        private struct ClassInformation
+        {
+            public string Name;
+            public string Namespace;
+            public string Guid;
+            public long FileId;
+        }
+
+        private const string YamlPrefix = "%YAML 1.1";
+
+        private static readonly Dictionary<string, string> sourceToOutputFolders = new Dictionary<string, string>
+        {
+            {"Library/PlayerDataCache/WindowsStoreApps", "UAPPlayer" },
+            {"Library/PlayerDataCache/Win", "StandalonePlayer" },
+        };
+
+        private static readonly HashSet<string> ExcludedYamlAssetExtensions = new HashSet<string> { ".jpg", ".csv", ".meta", ".pfx", ".txt", ".nuspec", ".asmdef", ".yml", ".cs", ".md", ".json", ".ttf", ".png", ".shader", ".wav", ".bin", ".gltf", ".glb", ".fbx", ".FBX", ".pdf", ".cginc" };
+        private static readonly HashSet<string> ExcludedSuffixFromCopy = new HashSet<string>() { ".cs", ".cs.meta", ".asmdef", ".asmdef.meta" };
+
+        private static Dictionary<string, string> nonClassDictionary = new Dictionary<string, string>(); //Guid, FileName
+
+        private const string ScriptFileIdConstant = "11500000";
```

Comment:
>private const string ScriptFileIdConstant = "11500000"; [](start = 8, length = 55)

Nit: Is this an arbitrarily chosen number/constant (or is this dependent on something else?)

Reply:
Added comment, it's the Unity defined script file id.

---
In reply to: [277822892](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277822892) [](ancestors = 277822892)